platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -59,135 +99,138 @@ export function ViewportGridProvider({ children, service }) {
         // which might have been a PDF Viewport. The viewport itself
         // will deal with inheritance if required. Here is just a simple
         // provider.
-        const viewportOptions = payload.viewportOptions || {};
-        const displaySetOptions = payload.displaySetOptions || [{}];
-
-        const viewports = state.viewports.slice();
+        const viewport = state.viewports[viewportIndex] || {};
+        const viewportOptions = {
+          ...viewport.viewportOptions,
+          ...payload.viewportOptions,
+        };
 
-        if (!viewportOptions.viewportId) {
-          viewportOptions.viewportId = `viewport-${viewportIndex}`;
+        // There can be display set options inheritance, since that is used
+        // to store reuseIds and other keys
```

Comment:
I am not sure what this comment refers to... in particular what reuseIds refers to. Please clarify.